conductor-client/src/main/kotlin/com/openlattice/assembler/PostgresRoles.kt

```diff
@@ -19,28 +17,31 @@ class PostgresRoles private constructor() {
     companion object {
 
         @JvmStatic
-        fun getOrCreatePermissionRole(
-                externalRoleNames: IMap<AccessTarget, UUID>,
-                tableId: UUID,
-                columnId: UUID,
-                permission: Permission
-        ): String {
-            return getOrCreatePermissionRole(externalRoleNames, AccessTarget.forPermissionOnTarget(permission, tableId, columnId))
-        }
-
-        @JvmStatic
-        fun getOrCreatePermissionRole( permissionRoleNames: IMap<AccessTarget, UUID>, column: TableColumn, permission: Permission): String {
-            return getOrCreatePermissionRole(permissionRoleNames, AccessTarget.forPermissionOnTarget(permission, column.tableId, column.columnId))
-        }
-
-        @JvmStatic
-        fun getOrCreatePermissionRole( permissionRoleNames: IMap<AccessTarget, UUID>, permission: Permission, root: UUID, vararg parts: UUID ): String {
-            return getOrCreatePermissionRole(permissionRoleNames, AccessTarget.forPermissionOnTarget(permission, root, *parts))
-        }
-
-        @JvmStatic
-        fun getOrCreatePermissionRole( permissionRoleNames: IMap<AccessTarget, UUID>, target: AccessTarget ): String {
-            return permissionRoleNames.executeOnKey(target, GetOrCreateExternalPermissionRoleNameEntryProcessor())
+        fun getOrCreatePermissionRolesAsync(
+                permissionRoleNames: IMap<AccessTarget, UUID>,
+                targets: Set<AccessTarget>,
+                createExternalDatabaseRole: (roleName: String) -> Unit
+        ): CompletionStage<Map<AccessTarget, UUID>> {
+            val allExistingRoleNames = permissionRoleNames.getAll(targets)
+            targets - allExistingRoleNames.keys
+            val targetsToNewIds = targets.filterNot {
+                // filter out roles that already exist
+                allExistingRoleNames.containsKey(it)
+            }.associateWith {
+                val newRole = UUID.randomUUID()
+                createExternalDatabaseRole(newRole.toString())
```

Comment:
I think this creates the permission role but doesn't actually grant any privileges to the role

Reply:
yes thats correct